include/libvfio-user.h

```diff
@@ -240,15 +240,16 @@ typedef ssize_t (vfu_region_access_cb_t) (void *pvt, char *buf, size_t count,
  * @flags: region  flags
  * @mmap_areas: array of memory mappable areas
  * @nr_mmap_areas: size of mmap_areas
- * @map: callback function to map region
+ * @fd: file descriptor of the file backing the region if it's a mappable
+ *      region
  *
```

Comment:
Can we get some more docs on how the mmap areas / fd is used? Just something describing that the client will be able to request the given fd, and will mmap the regions based upon the sparse mappings.

Maybe we should also point out that there's nothing stopping the client mapping anything in that fd regardless of the areas passed in. Or maybe that's too obvious.

Reply:
Sure, will elaborate.